src/main/java/com/google/starfish/servlets/HandleNotesServlet.java

```diff
@@ -1,58 +1,128 @@
-package com.google.servlets;
+package com.google.starfish.servlets;
 
 import com.google.appengine.api.blobstore.BlobInfo;
 import com.google.appengine.api.blobstore.BlobInfoFactory;
 import com.google.appengine.api.blobstore.BlobKey;
 import com.google.appengine.api.blobstore.BlobstoreService;
 import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
-import com.google.appengine.api.blobstore.BlobInfoFactory;
-import com.google.appengine.api.blobstore.BlobInfo;
-import com.google.appengine.api.images.ServingUrlOptions;
 import java.io.IOException;
-import java.io.PrintWriter;
-import java.net.MalformedURLException;
-import java.net.URL;
 import java.util.List;
 import java.util.Map;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;  
 import javax.servlet.http.Cookie;  
-
+import com.google.starfish.services.LabelService;
+import com.google.starfish.services.MiscNoteLabelService;
+import javax.sql.DataSource;
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Date;
+import java.util.Calendar;
+
+/** Servlet to handle note uploading and adding associated labels to the database */
 @WebServlet("/handle-notes")
 public class HandleNotesServlet extends HttpServlet {
 
   private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
   private final String COOKIE_NAME = "SFCookie";
+  private LabelService labelService = new LabelService();
+  private MiscNoteLabelService miscNoteLabelService = new MiscNoteLabelService();
   
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
     if(!validateUser(request)) {
       response.setStatus(HttpServletResponse.SC_FORBIDDEN);
       return;
     }
 
     HttpSession activeSession = request.getSession(false);
     String userId = (String) activeSession.getAttribute("user_id");
-
     String blobKey = getUploadedFileBlobKey(request, "file");
-    if(blobKey == null) {
+    String sourceUrl = request.getParameter("sourceUrl");
+
+    // There must be some source, either a blob or url, for this note to be posted
+    if (blobKey == null && sourceUrl == null) {
       response.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
       return;
     }
-    String title = request.getParameter("title").toLowerCase();
+
+    String title = request.getParameter("title");
     String school = request.getParameter("school").toLowerCase();
     String course = request.getParameter("course").toLowerCase();
     String[] miscLabels = request.getParameter("miscLabels").split(",");
```

Comment:
I should probably make sure the front end doesn't allow commas in label names, then. But how about [getParameterValues](https://www.java4s.com/java-servlet-tutorials/example-on-getparametervalues-method-of-servlet-request/), instead? This is a bug waiting to happen, ATM.

Reply:
I tried testing getParameterValues once, and it seemed to just combine all the labels together. I'm not sure if I was doing it wrong, but maybe we could try it again.

Reply:
From what I tested, `.split(",")` seemed to be working ok but I see how this could easily become a bug. `getParameterValues` definitely looks like the right solution here so I'll make that change, good catch!

Reply:
@keelywan sounds like maybe you did

`String values = getParameterValues(“Input Parameter”);`

instead of

`String[] values = getParameterValues(“Input Parameter”);` 
?